### PR TITLE
Zed pulp & package repo config

### PR DESCRIPTION
**Pull request by Alex-Welsh**
This PR mostly contains config changes for Pulp and Zed-specific package repos (again, cleaving out a bunch of CS8/RL8/focal stuff).

There are also some commits for kolla container image building, since many of the changes were made at the same time. Zed container image builds still require some more changes after these, but this PR is already plenty big enough.

### CODE REVIEW

**Comment by Alex-Welsh** [comment]
I can try to subdivide this further if it would help

**Comment by markgoddard** [comment]
> I can try to subdivide this further if it would help

I can handle it :D

File: etc/kayobe/apt.yml [hunk 1]
```diff
@@ -63,12 +63,12 @@ stackhpc_apt_repositories:
...
 
 # Disable pulp apt repos on Ubuntu Jammy until they are published.
-apt_repositories: "{{ [] if ansible_facts['distribution_release'] == 'jammy' else stackhpc_apt_repositories }}"
+apt_repositories: "{{ stackhpc_apt_repositories }}"
 
 # Whether to disable repositories in /etc/apt/sources.list. This may be used
 # when replacing the distribution repositories via apt_repositories.
 # Default is false.
-apt_disable_sources_list: "{{ ansible_facts['distribution_release'] != 'jammy' }}"
+apt_disable_sources_list: "true"
```
Comment (markgoddard) [inline etc/kayobe/apt.yml]:
```suggestion
apt_disable_sources_list: true
```

File: etc/kayobe/environments/ci-aio/kolla/globals.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ nova_compute_virt_type: qemu
 openstack_service_workers: "1"
 openstack_service_rpc_workers: "1"
 
-docker_yum_baseurl: "{{ stackhpc_repo_docker_url }}"
+docker_yum_baseurl: "{{ stackhpc_repo_centos_stream_9_docker_url }}"
 docker_yum_gpgkey: "https://download.docker.com/linux/centos/gpg"
```
Comment (markgoddard) [inline etc/kayobe/environments/ci-aio/kolla/globals.yml]: I don't think we need these since docker is in dnf.yml

File: etc/kayobe/kolla.yml [hunk 3]
```diff
@@ -4,9 +4,6 @@
 ###############################################################################
 # Kolla installation.
 
-# Type of Kolla control installation. One of 'binary' or 'source'.
-#kolla_ctl_install_type:
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: This refers to the installation of the kolla python package, not the packages in containers.

File: etc/kayobe/kolla.yml [hunk 4]
```diff
@@ -66,8 +63,8 @@ kolla_ansible_source_version: "{{ stackhpc_kolla_ansible_source_version }}"
 ###############################################################################
 # Kolla configuration.
 
-# Kolla base container image distribution. Options are "centos", "debian",
-# "rocky", "ubuntu". Default is {{ os_distribution }}.
+# Kolla base container image distribution. Options are "rocky", "ubuntu".
+# Default is {{ os_distribution }}.
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: Probably best to stick with upstream for these or we'll end up in merge hell.

File: etc/kayobe/kolla.yml [hunk 5]
```diff
@@ -336,16 +272,6 @@ kolla_build_blocks:
         rm -f liblasso3_2.7.0-2build3_amd64.deb
     {% endif %}
     {% endraw %}
-  nova_base_footer: |
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml:335]: @cityofships do you know/remember whether the liblasso fix is still necessary on Jammy?

File: etc/kayobe/kolla.yml [hunk 6]
```diff
@@ -360,7 +286,7 @@ kolla_build_customizations_common:
   ironic_inspector_pip_packages_append:
     - /additions/*
 
-kolla_build_customizations_centos:
+kolla_build_customizations_rocky: # TODO: Check whether this is actually correct. It seems that it was previously being skipped for rocky
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml:288]: @mnasiadka are the OVN/OVS bits still relevant on RL9?

File: etc/kayobe/pulp.yml [hunk 7]
```diff
@@ -226,232 +208,7 @@ stackhpc_pulp_repo_rocky_9_highavailability_version: "{{ lookup('vars', 'stackhp
...
-    remote_password: "{{ stackhpc_release_pulp_password }}"
-    client_cert: ""
-    client_key: ""
-    policy: on_demand
-    proxy_url: "{{ pulp_proxy_url }}"
-    sync_policy: mirror_complete
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_8 | bool }}"
-
-  - name: Grafana
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml:431]: Looks like we've lost Grafana.

File: etc/kayobe/pulp.yml [hunk 8]
```diff
@@ -621,6 +366,24 @@ stackhpc_pulp_repository_rpm_repos:
     sync_policy: mirror_complete
     state: present
     required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: OpenSearch for RHEL 9
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml]:
```suggestion
  - name: OpenSearch 2.x
```

File: etc/kayobe/pulp.yml [hunk 9]
```diff
@@ -621,6 +366,24 @@ stackhpc_pulp_repository_rpm_repos:
...
+    url: "{{ stackhpc_release_pulp_content_url }}/pulp/content/opensearch/2.x/yum/{{ stackhpc_pulp_repo_opensearch_2_x_version }}"
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml]:
```suggestion
    url: "{{ stackhpc_release_pulp_content_url }}/opensearch/2.x/yum/{{ stackhpc_pulp_repo_opensearch_2_x_version }}"
```

File: etc/kayobe/pulp.yml [hunk 10]
```diff
@@ -621,6 +366,24 @@ stackhpc_pulp_repository_rpm_repos:
...
+    remote_username: "{{ stackhpc_release_pulp_username }}"
+    remote_password: "{{ stackhpc_release_pulp_password }}"
+    policy: on_demand
+    proxy_url: "{{ pulp_proxy_url }}"
+    sync_policy: mirror_complete
+    state: present
+    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: OpenSearch Dashboards for RHEL 9
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml]:
```suggestion
  - name: OpenSearch Dashboards 2.x
```

File: etc/kayobe/pulp.yml [hunk 11]
```diff
@@ -621,6 +366,24 @@ stackhpc_pulp_repository_rpm_repos:
...
+    url: "{{ stackhpc_release_pulp_content_url }}/pulp/content/opensearch-dashboards/2.x/yum/{{ stackhpc_pulp_repo_opensearch_dashboards_2_x_version }}"
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml]:
```suggestion
    url: "{{ stackhpc_release_pulp_content_url }}/opensearch-dashboards/2.x/yum/{{ stackhpc_pulp_repo_opensearch_dashboards_2_x_version }}"
```

File: etc/kayobe/pulp.yml [hunk 12]
```diff
@@ -631,125 +394,17 @@ stackhpc_pulp_publication_rpm_development: "{{ stackhpc_pulp_distribution_rpm_de
...
-    base_path: "centos/8-stream/storage/x86_64/ceph-pacific/development"
-    repository: CentOS Stream 8 - Ceph Pacific
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_8 | bool }}"
-  - name: "elasticsearch-logstash-kibana-7.x-development"
-    base_path: "elasticsearch/oss-7.x/development"
-    repository: ELK repository for 7.x packages
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_8 | bool }}"
-  - name: "grafana-development"
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml:716]: ditto

File: etc/kayobe/pulp.yml [hunk 13]
```diff
@@ -826,129 +481,31 @@ stackhpc_pulp_distribution_rpm_development:
...
-    state: present
-    required: "{{ stackhpc_pulp_sync_rocky_8 | bool }}"
-
-  # EPEL repositories
-  - name: "extra-packages-for-enterprise-linux-8-x86_64-production"
-    base_path: "epel/8/Everything/x86_64/production"
-    distribution: "extra-packages-for-enterprise-linux-8-x86_64-development"
+  - name: "opensearch-2-x-development"
+    base_path: "opensearch/2.x/yum/development"
+    repository: OpenSearch for RHEL 9
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml]:
```suggestion
    repository: OpenSearch 2.x

File: etc/kayobe/pulp.yml [hunk 14]
```diff
@@ -826,129 +481,31 @@ stackhpc_pulp_distribution_rpm_development:
...
     state: present
-    required: "{{ stackhpc_pulp_sync_el_8 | bool }}"
-  - name: "extra-packages-for-enterprise-linux-modular-8-x86_64-production"
-    base_path: "epel/8/Modular/x86_64/production"
-    distribution: "extra-packages-for-enterprise-linux-modular-8-x86_64-development"
+    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: "opensearch-dashboards-2-x-development"
+    base_path: "opensearch-dashboards/2.x/yum/development"
+    repository: OpenSearch for RHEL 9
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml]:
```suggestion
    repository: OpenSearch Dashboards 2.x

File: etc/kayobe/pulp.yml [hunk 15]
```diff
@@ -826,129 +481,31 @@ stackhpc_pulp_distribution_rpm_development:
...
-    base_path: "centos/8-stream/storage/x86_64/ceph-pacific/production"
-    repository: CentOS Stream 8 - Ceph Pacific
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_8 | bool }}"
-  - name: "elasticsearch-logstash-kibana-7.x-production"
-    base_path: "elasticsearch/oss-7.x/production"
-    repository: ELK repository for 7.x packages
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_8 | bool }}"
-  - name: "grafana-production"
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml:910]: ditto

File: etc/kayobe/pulp.yml [hunk 16]
```diff
@@ -1025,6 +582,16 @@ stackhpc_pulp_distribution_rpm_production:
     repository: InfluxDB for RHEL 9
     state: present
     required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: "opensearch-2-x-production"
+    base_path: "opensearch/2.x/yum/production"
+    repository: OpenSearch for RHEL 9
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml]:
```suggestion
    repository: OpenSearch 2.x
```

File: etc/kayobe/pulp.yml [hunk 17]
```diff
@@ -1025,6 +582,16 @@ stackhpc_pulp_distribution_rpm_production:
...
+    state: present
+    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: "opensearch-dashboards-2-x-production"
+    base_path: "opensearch-dashboards/2.x/yum/production"
+    repository: OpenSearch for RHEL 9
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml]:
```suggestion
    repository: OpenSearch Dashboards 2.x
```

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: etc/kayobe/pulp.yml (hunk 7 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml:431]: Ah, yes. I think this is one of those things I missed then fixed in a later commit. I'll correct it here

File: etc/kayobe/pulp.yml (hunk 12 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml:716]: Interesting that this wasn't being synced for RL9 to begin with. Is it necessary?

File: etc/kayobe/apt.yml (hunk 1 above)
Comment (Alex-Welsh) [reply etc/kayobe/apt.yml]: Done

File: etc/kayobe/environments/ci-aio/kolla/globals.yml (hunk 2 above)
Comment (Alex-Welsh) [reply etc/kayobe/environments/ci-aio/kolla/globals.yml]: Removed

File: etc/kayobe/kolla.yml (hunk 3 above)
Comment (Alex-Welsh) [reply etc/kayobe/kolla.yml]: Odd this has cropped up twice. Reverted

File: etc/kayobe/kolla.yml (hunk 4 above)
Comment (Alex-Welsh) [reply etc/kayobe/kolla.yml]: Good point. Reverted

File: etc/kayobe/pulp.yml (hunk 7 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml:431]: Fixed

File: etc/kayobe/pulp.yml (hunk 8 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml]: Done

File: etc/kayobe/pulp.yml (hunk 9 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml]: Done

File: etc/kayobe/pulp.yml (hunk 10 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml]: Done

File: etc/kayobe/pulp.yml (hunk 11 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml]: Done

File: etc/kayobe/pulp.yml (hunk 12 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml:716]: Fixed

File: etc/kayobe/pulp.yml (hunk 13 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml]: Done

File: etc/kayobe/pulp.yml (hunk 15 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml:910]: Fixed

File: etc/kayobe/pulp.yml (hunk 14 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml]: Done

File: etc/kayobe/pulp.yml (hunk 17 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml]: Done

File: etc/kayobe/pulp.yml (hunk 16 above)
Comment (Alex-Welsh) [reply etc/kayobe/pulp.yml]: Done

File: etc/kayobe/pulp.yml (hunk 12 above)
Comment (markgoddard) [reply etc/kayobe/pulp.yml:716]: https://github.com/stackhpc/stackhpc-kayobe-config/pull/605/files

**Comment by markgoddard** [comment]
Has some merge conflicts

**Comment by Alex-Welsh** [comment]
> Has some merge conflicts

Rebased & resolved

**Review by markgoddard — COMMENTED** [review]
LGTM pending open questions for @cityofships and @mnasiadka

File: etc/kayobe/kolla.yml (hunk 6 above)
Comment (mnasiadka) [reply etc/kayobe/kolla.yml:288]: If we won't pin to OVS 2.17 and OVN whatever is latest, we might get OVS 3.1 - which is not LTS, but still should work (and give us some new features). I think it should be some joint decision if we are going OVS 3.1 path or not.
@bbezak @oneswig any insights?

Comment (bbezak) [reply etc/kayobe/kolla.yml:288]: I'm happy to use ovs 3.1 - however we need to test that before deploying to prod branches

Comment (markgoddard) [reply etc/kayobe/kolla.yml:288]: We're not applying these build customisations to RL9 on Yoga. Does that mean we may get OVS 3.1 there too?

File: etc/kayobe/kolla.yml (hunk 5 above)
Comment (Alex-Welsh) [reply etc/kayobe/kolla.yml:335]: I don't have the cherry pick here but I did actually have to remove this in the zed-init branch to get keystone to build https://github.com/stackhpc/stackhpc-kayobe-config/commit/d84a6214505d70c4eb3bae5d2a3bda339fcade44

Comment (Alex-Welsh) [reply etc/kayobe/kolla.yml:335]: I was going to add it to a final _kolla image build fixes_ PR but can put it in here if you want

Comment (markgoddard) [reply etc/kayobe/kolla.yml:335]: That's fine, as long as it goes in eventually.

**Review by markgoddard — APPROVED** [review]

File: etc/kayobe/kolla.yml (hunk 6 above)
Comment (mnasiadka) [reply etc/kayobe/kolla.yml:288]: Well, for now it's OVS 2.17 and OVN 22.09 - but that's due to pinning in virtual rdo-openvswitch and rdo-ovn packages, but RDO is in control now - not us ;-)

**Pull request merged**